src/lib/cli.rs

```diff
@@ -131,47 +137,37 @@ fn endpoints_parser(entry: &str) -> Result<Arc<dyn drivers::Driver>, String> {
     drivers::create_driver_from_entry(entry)
 }
 
-struct Manager {
-    clap_matches: Args,
-}
-
-lazy_static! {
-    static ref MANAGER: Arc<Manager> = Arc::new(Manager::new());
-}
-
-impl Manager {
-    fn new() -> Self {
-        Self {
-            clap_matches: Args::parse(),
-        }
-    }
+/// Constructs our manager, Should be done inside main
+#[instrument(level = "debug")]
+pub fn init() {
+    init_with(Args::parse());
 }
 
 /// Constructs our manager, Should be done inside main
-
 #[instrument(level = "debug")]
-pub fn init() {
-    MANAGER.as_ref();
+pub fn init_with(args: Args) {
+    MANAGER.get_or_init(|| Manager { clap_matches: args });
 }
 
 /// Checks if the verbosity parameter was used
-
 #[instrument(level = "debug")]
 pub fn is_verbose() -> bool {
-    MANAGER.clap_matches.verbose
+    MANAGER.get().unwrap().clap_matches.verbose
 }
 
 #[instrument(level = "debug")]
 pub fn is_tracing() -> bool {
-    MANAGER.clap_matches.enable_tracing_level_log_file
+    MANAGER
+        .get()
+        .unwrap()
+        .clap_matches
+        .enable_tracing_level_log_file
 }
 
-/// Our log path
-
 #[instrument(level = "debug")]
 pub fn log_path() -> String {
     let log_path =
-        MANAGER.clap_matches.log_path.clone().expect(
+        MANAGER.get().unwrap().clap_matches.log_path.clone().expect(
```

Comment:
can we have a macro to make the access easier ?

Reply:
I made it into a local function instead =)